demos/http/http_demo_plaintext/http_demo_plaintext.c

```diff
@@ -0,0 +1,552 @@
+#include <assert.h>
+#include <stdlib.h>
+#include <string.h>
+
+/* POSIX socket includes. */
+#include <errno.h>
+#include <netdb.h>
+#include <time.h>
+#include <unistd.h>
+#include <arpa/inet.h>
+
+#include <sys/socket.h>
+#include <sys/types.h>
+
+/* HTTP API header. */
+#include "http_client.h"
+
+/* Demo config header. */
+#include "demo_config.h"
+
+/**
+ * @brief HTTP server host name.
+ *
+ * This demo uses httpbin.org: A simple HTTP Request & Response Service.
+ */
+#define SERVER_HOST    "httpbin.org"
+
+/**
+ * @brief HTTP server port number.
+ *
+ * In general, port 80 is for plaintext HTTP connections.
+ */
+#define SERVER_PORT    80
+
+/**
+ * @brief Paths for different HTTP methods for specified host.
+ *
+ * For httpbin.org, see http://httpbin.org/#/HTTP_Methods for details on
+ * supported REST API.
+ */
+#ifndef GET_PATH
+    #define GET_PATH    "/get"
+#endif
+
+#ifndef HEAD_PATH
+    #define HEAD_PATH    "/get"
+#endif
+
+#ifndef PUT_PATH
+    #define PUT_PATH    "/put"
+#endif
+
+#ifndef POST_PATH
+    #define POST_PATH    "/post"
+#endif
+
+/**
+ * @brief Transport timeout in milliseconds for transport send and receive.
+ */
+#define TRANSPORT_SEND_RECV_TIMEOUT_MS    ( 1000 )
+
+/**
+ * @brief The length in bytes of the user buffer.
+ */
+#define USER_BUFFER_LENGTH                ( 1024 )
+
+/**
+ * @brief Length of an IPv6 address when converted to hex digits.
+ */
+#define IPV6_LENGTH                       ( 40 )
+
+
+/**
+ * @brief Some text to send as the request body for PUT and POST requests in
+ * this demo.
+ */
+#define REQUEST_BODY_TEXT           "Hello, world!"
+#define REQUEST_BODY_TEXT_LENGTH    ( sizeof( REQUEST_BODY_TEXT ) - 1 )
+
+/**
+ * @brief A string to store the resolved IP address from the host name.
+ */
+static char resolvedIpAddr[ IPV6_LENGTH ] = { 0 };
+
+/**
+ * @brief A buffer used to store request headers and reused after sending
+ * the request to store the response headers and body.
+ *
+ * User can also decide to have two separate buffers for request and response.
+ */
+static uint8_t userBuffer[ USER_BUFFER_LENGTH ] = { 0 };
+
+/**
+ * @brief The request body.
+ */
+static uint8_t requestBodyBuffer[ REQUEST_BODY_TEXT_LENGTH ] = { 0 };
+
+/**
+ * @brief Definition of the HTTP network context.
+ */
+struct HTTPNetworkContext
+{
+    int tcpSocket;
+};
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Performs a DNS lookup on the given host name then establishes a TCP
+ * connection to the server.
+ *
+ * @param[in] pServer Host name of server.
+ * @param[in] port Server port.
+ * @param[out] pTcpSocket Pointer to TCP socket file descriptor.
+ *
+ * @return EXIT_FAILURE on failure; EXIT_SUCCESS on success.
+ */
+static int connectToServer( const char * pServer,
+                            uint16_t port,
+                            int * pTcpSocket );
+
+/**
+ * @brief The transport send function that defines the transport interface.
```

Comment:
(Nit) "The definition of the `send()` function in the transport interface. It is used for sending packets over the TCP transport layer."

The above description can be updated to mention about TLS in the other HTTP demos.